### DISCUSSION TITLE
Show discard alert on view will disappear

### DISCUSSION THREAD

**Discussion by elyahk** [pointfreeco/swift-composable-architecture#2387] [Q&A]
Hi guys, I am migrating my project to SwiftUI with Composable architecture, on my UIkit project there are discard changes alerts when you edit something and return back without saving changes. Here I will show the discard changes alert. But in SwiftUI and Composable architecture, I couldn't find a solution to do it. Any ideas?

**Comment by stephencelis** [comment]
Hi @elyahk! We're not super familiar with how UIKit discard change alerts are set up and configured, but I would first try to figure out how to achieve these same alerts in vanilla SwiftUI before trying to achieve the same in TCA.

Also, do you want to share some more code so that folks have something to work with to help out? Maybe some code showing how it works in UIKit, and then some SwiftUI code that you'd like to behave similarly?
